sdv/sampler.py

```diff
@@ -48,6 +63,15 @@ def transform_synthesized_rows(self, synthesized_rows, table_name, num_rows):
         meta = self.dn.tables[table_name].meta
         orig_meta = self._get_table_meta(self.dn.meta, table_name)
         primary_key = meta.get('primary_key')
+        categorical_fields = [
```

Comment:
Instead of:
* look over the fields list once here
* evaluate an if for each one of them
* evaluate an if to see if there are any categorical fields
* loop over the categorical fields again

I would loop just once and have an "if categorical: check values and rescale if necessary" inside the loop.